DNS Probe

pchaigno opened the pull request:
I made a first quick implementation for a DNS probe. I'd like to clean it up and upstream it (that'd be my first Go implementation ever :smile: ), but first, I want to see with you how you think it should be exposed to the user.

It currently is exposed as:

``` go
func NewDNS(addrport string, warning, fatal time.Duration) *DNS
```

Should I also allow the user to change the tested domain (currently `A google.com`) and check the answer? Should we really expose the destination port as well?

Also, I use this Go library for the DNS client: https://github.com/miekg/dns. Do you have a better suggestion or is that fine?

Lesterpig commented:
This Go library seems pretty cool, good choice!

In an ideal world, it would be great to customize the DNS to make. Is it a very simple way to provide a fourth parameter (string?) to the NewDNS function with a custom DNS request?

For example `MX mailserver.me` or `AAAA myipv6.blog`

I don't really know for the port, maybe we can remove it for the DNS case, it's mostly up to you :smile: 

Additionnaly, it would be great to check the DNS response in a simple way, if I want to check that `A my.blog` is `1.2.3.4` and not corrupted by someone. What do you think about this?

So, the ideal prototype would be:

``` go
func NewDNS(addr, query, expected string, warning, fatal time.Duration) *DNS
```

Thank you for your work on this :)

pchaigno committed:
DNS probe

pchaigno commented:
I converted this issue into a pull request with the required changes. I've also updated the README and the Dockerfile.

It only allows you to perform DNS queries of type A. I don't know if it's worth enabling other types of queries.

Lesterpig commented:
Fine, thanks a lot!

I'll merge it as-is, but'll probably fix these _really_ small issues later :stuck_out_tongue_winking_eye: 
- `go get` downloads and installs the specified module, no need to build it after
- the code is not formatted with go standards. Be lazy and use gofmt (included in go):

``` bash
$ gofmt -d . # see diff suggested by go
$ gofmt -w . # let go reformat your files for you
```

pchaigno commented:
Good to know. Don't hesitate to tell me before merging next time if I miss it. Style consistency is important.
